transport/tchannel/config.go

```diff
@@ -32,11 +33,24 @@ import (
 
 const transportName = "tchannel"
 
+var defaultConnectionTimeout = 500 * time.Millisecond
+
 // TransportConfig configures a shared TChannel transport. This is shared
 // between all TChannel outbounds and inbounds of a Dispatcher.
 //
-// TransportConfig does not have any parameters at this time.
-type TransportConfig struct{}
+//  transports:
+//    tchannel:
+//      connection-timeout: 500ms
+type TransportConfig struct {
+	ConnectionTimeout time.Duration `config:"connection-timeout,interpolate"`
+}
+
+func newTransportConfig() transportConfig {
```

Comment:
Stack vs heap. This way there’s a memcopy off the stack, no alloc. In this case, it’s six of one, half dozen of the other.